lib/codegen/src/verifier/mod.rs

```diff
@@ -82,23 +82,86 @@ pub use self::cssa::verify_cssa;
 pub use self::liveness::verify_liveness;
 pub use self::locations::verify_locations;
 
-// Create an `Err` variant of `VerifierResult<X>` from a location and `format!` arguments.
-macro_rules! err {
-    ( $loc:expr, $msg:expr ) => {
-        Err(::verifier::VerifierError {
+/// Report an error.
+///
+/// The first argument must be a `&mut VerifierErrors` reference, and the following
+/// argument defines the location of the error and must implement `Into<AnyEntity>`.
+/// Finally, subsequent arguments will be formatted using `format!()` and set
+/// as the error message.
+macro_rules! report {
+    ( $errors: expr, $loc: expr, $msg: tt ) => {
+        $errors.0.push(::verifier::VerifierError {
             location: $loc.into(),
             message: String::from($msg),
         })
     };
 
-    ( $loc:expr, $fmt:expr, $( $arg:expr ),+ ) => {
-        Err(::verifier::VerifierError {
+    ( $errors: expr, $loc: expr, $fmt: tt, $( $arg: expr ),+ ) => {
+        $errors.0.push(::verifier::VerifierError {
             location: $loc.into(),
             message: format!( $fmt, $( $arg ),+ ),
         })
     };
 }
 
+/// Diagnose a fatal error, and return `Err`.
+macro_rules! fatal {
+    ( $( $arg: expr ),+ ) => ({
+        report!( $( $arg ),+ );
+        Err(())
+    });
+}
+
+/// Diagnose a non-fatal error, and return `Ok`.
+macro_rules! nonfatal {
+    ( $( $arg: expr ),+ ) => ({
+        report!( $( $arg ),+ );
+        Ok(())
+    });
+}
+
+/// Faster syntax for calling `verify_*` functions.
```

Comment:
"Shorthand" would convey the purpose here better than "faster". Also, a good one-line summary of this macro is that it adapts a function returning a `VerifierStepResult` to return a `VerifierResult`.